test/gurka/test_gtfs.cc

```diff
@@ -146,4 +160,94 @@ TEST(GtfsExample, WriteGtfs) {
+            gtfs::CalendarDateException::Removed); // service added
+
+  const auto& frequencies = feed_reader.get_frequencies();
+  EXPECT_EQ(frequencies.size(), 2);
+  EXPECT_EQ(frequencies[0].exact_times, gtfs::FrequencyTripService::FrequencyBased);
+  EXPECT_EQ(frequencies[1].exact_times, gtfs::FrequencyTripService::ScheduleBased);
+}
+
+TEST(GtfsExample, MakeTiles) {
+  filesystem::create_directories("test/data/transit_test");
+  filesystem::create_directories("test/data/transit_tiles");
+  auto pt = test::make_config("test/data/transit_test",
+                              {{"mjolnir.transit_feeds_dir", "test/data/gtfs_feeds/"},
+                               {"mjolnir.transit_dir", "test/data/transit_tiles"}});
+
+  // constants written in the last function
+  const std::string tripOneID = "10";
+  const std::string tripTwoID = "11";
+  const std::string stopOneID = "7";
+  const std::string stopTwoID = "8";
+  const std::string shapeOneID = "5";
+  const std::string serviceOneID = "9";
+  const int serviceStartDate = 20220131;
+  const int serviceEndDate = 20230131;
+  const int addedDate = 20220202;
+  const int removedDate = 20220203;
+  const int headwaySec = 1800;
+
+  // spawn threads to download all the tiles returning a list of
+  // tiles that ended up having dangling stop pairs
+  auto dangling_tiles = valhalla::mjolnir::ingest_transit(pt);
+
+  // spawn threads to connect dangling stop pairs to adjacent tiles' stops
+  valhalla::mjolnir::stitch_transit(pt, dangling_tiles);
+  // call the two functions, in main valhalla_ingest-transit
+  // it's gonna write protobufs
+  filesystem::recursive_directory_iterator transit_file_itr("test/data/transit_tiles");
+  filesystem::recursive_directory_iterator end_file_itr;
+
+  // for each pbf.
+  for (; transit_file_itr != end_file_itr; ++transit_file_itr) {
+    if (filesystem::is_regular_file(transit_file_itr->path())) {
+      std::string fname = transit_file_itr->path().string();
+      mjolnir::Transit transit = mjolnir::read_pbf(fname);
+
+      // make sure we are looking at a pbf file
+
+      // make sure that the data written in the previous test is readable through pbfs
+      // shape info
+      // becomes 1 shape from 4 shape_points last test
+      EXPECT_EQ(transit.shapes_size(), 1);
+      EXPECT_EQ(transit.shapes(0).shape_id(), stoi(shapeOneID));
+      // stop(node) info
+      EXPECT_EQ(transit.nodes_size(), 2);
+      bool stops_written =
+          transit.nodes(0).onestop_id() == stopOneID || transit.nodes(0).onestop_id() == stopTwoID;
+      EXPECT_EQ(stops_written, true);
+      // routes info
+      EXPECT_EQ(transit.routes_size(), 1);
+      EXPECT_EQ(transit.routes(0).onestop_id(), "2");
+
+      // stop_pair info
+      EXPECT_EQ(transit.stop_pairs_size(), 2);
+      bool stop_pairs_written = transit.stop_pairs(0).origin_onestop_id() == stopOneID ||
+                                transit.stop_pairs(0).origin_onestop_id() == stopTwoID;
```

Comment:
does the code coverage warning about not hitting line 233 mean that the id is always stopOneID and never stopTwoID?

Reply:
Yup, this is expected - stopTwoID should be stored in `transit.stop_paris(1)`. 
However, `transit.stop_pairs(0) = stopTwoID` on my device, which made me realize that depending on threading, it might store the data in a different order. 
But on the same device, the order should be the same.

Reply:
oh jeez for some reason i didnt see the `(0)` i thought that the loop just checked both but i see you check 0 and 1 separately in the loop, sorry about that!

Reply:
actually it should be reproducable, if we set concurrency to 1 in the config (which i think it is..) only one thread will ever be there to pick of stuff to do in order. we dont randomize anything else on purpose do we?

Reply:
No, that is all for checking for randomness